sqlx-sqlite/src/connection/explain.rs

```diff
@@ -198,45 +198,171 @@ impl RegDataType {
     }
 }
 
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug, Clone, Eq, Default)]
+struct IntMap<V: std::fmt::Debug + Clone + Eq + PartialEq + std::hash::Hash>(Vec<Option<V>>);
```

Comment:
I'm not really thrilled that this VecMap struct actually gave a speedup, since I'd prefer sticking to standard containers.
But unlike HashMap, it implements Hash (removing the need for BranchStateHash).
And unlike Vec, it allows populating indexes out-of-order without thinking about length.

I also experimented with BTreeMap (to get Hash-able), but BTreeMap was slightly slower than the original implementation.

All this to say, I think the later FK-elision commit gives a more impactful speedup.  So if anyone else has concerns with this, I'll just drop this particular commit from the PR.

Reply:
I think this is fine, but I think it should live in its own module since it adds 100+ lines to an already very dense file.

I would also say to address the dead code warnings, either by deleting the unused methods or marking them `#[allow(dead_code)]` if you think they're worth keeping around.